src/cpp/rtps/writer/StatefulPersistentWriter.cpp

```diff
@@ -51,12 +55,30 @@ void StatefulPersistentWriter::unsent_change_added_to_history(
     StatefulWriter::unsent_change_added_to_history(cptr, max_blocking_time);
 }
 
-bool StatefulPersistentWriter::change_removed_by_history(CacheChange_t* change)
+bool StatefulPersistentWriter::change_removed_by_history(
+        CacheChange_t* change)
 {
     remove_persistent_change(change);
     return StatefulWriter::change_removed_by_history(change);
 }
 
+void StatefulPersistentWriter::print_inconsistent_acknack(
+        const GUID_t& writer_guid,
+        const GUID_t& reader_guid,
+        const SequenceNumber_t& min_requested_sequence_number,
+        const SequenceNumber_t& max_requested_sequence_number,
+        const SequenceNumber_t& next_sequence_number)
+{
+    if (!log_error_printed_)
+    {
+        log_error_printed_ = true;
+        logError(RTPS_WRITER, "Inconsistent acknack received in Local Writer "
+                << writer_guid << ". Maybe the persistent database has been erased locally.");
```

Comment:
If the database has been erased locally, does the user need to do something? If so, I'd suggest the actions explicitly:
```suggestion
        logError(RTPS_WRITER, "Inconsistent acknack received in Local Writer "
                << writer_guid << ". If the persistent database has been erased locally, <do whatever needs to be done>.");
```
If there is no action to be taken by the user, and this is just a log to record the possibility, ignore this comment.

Reply:
👍